src/platform/tizen/iotjs_module_i2c-tizen.c

```diff
@@ -0,0 +1,131 @@
+ * to deal in the Software without restriction, including without limitation
+ * the rights to use, copy, modify, merge, publish, distribute, sublicense,
+ * and/or sell copies of the Software, and to permit persons to whom
+ * the Software is furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+ * THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+ * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+ * DEALINGS IN THE SOFTWARE.
+ */
+
+/* Copyright 2017-present Samsung Electronics Co., Ltd. and other contributors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+
+/* Some functions are modified from the RoadNarrows-robotics i2c library.
+ * (distributed under the MIT license.)
+ */
+
+
+#include <fcntl.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/ioctl.h>
+#include <sys/stat.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <peripheral_io.h>
+
+#include "modules/iotjs_module_i2c.h"
+
+
+#define I2C_WORKER_INIT_TEMPLATE                                            \
+  iotjs_i2c_reqwrap_t* req_wrap = iotjs_i2c_reqwrap_from_request(work_req); \
+  iotjs_i2c_reqdata_t* req_data = iotjs_i2c_reqwrap_data(req_wrap);
+
+
+// The address can be set just once.
+void I2cSetAddress(iotjs_i2c_t* i2c, uint8_t address) {
+  IOTJS_VALIDATED_STRUCT_METHOD(iotjs_i2c_t, i2c);
+
+  if (_this->address == -1) {
+    // Perform delayed construction.
+    _this->address = address;
+    if (peripheral_i2c_open(_this->bus, _this->address, &_this->handle) < 0) {
+      // TODO: report the error at this point
+    }
+  } else if (_this->address != address) {
+    // TODO: report error OR recreate the object with new slave address
+  }
+}
+
+void OpenWorker(uv_work_t* work_req) {
+  I2C_WORKER_INIT_TEMPLATE;
+  // Not used for the deferred construction.
+  // iotjs_i2c_t* i2c = iotjs_i2c_instance_from_reqwrap(req_wrap);
+  // IOTJS_VALIDATED_STRUCT_METHOD(iotjs_i2c_t, i2c);
```

Comment:
Disagree. The deferred initialization is not usually done in iotjs HW components, so when it is done, at least some explanation should be in place.